godel_simple_gui/src/states/executing_state.cpp

```diff
@@ -64,23 +59,13 @@ void godel_simple_gui::ExecutingState::executeAll(BlendingWidget& gui)
 
 void godel_simple_gui::ExecutingState::executeOne(const std::string& plan, BlendingWidget& gui)
 {
-  /*
-  ROS_DEBUG_STREAM("Executing " << plan);
-  godel_msgs::SelectMotionPlan srv;
-  srv.request.name = plan;
-  srv.request.simulate = false;
-  srv.request.wait_for_execution = true;
-  if (!real_client_.call(srv))
-  {
-    std::ostringstream ss;
-    ss << "Failed to execute plan: " << plan << ". Please see logs for more details.";
-    throw BadExecutionError(ss.str());
-  }
-  */
-  ROS_INFO_STREAM("In select motion plan");
   godel_msgs::SelectMotionPlanActionGoal goal;
   goal.goal.name = plan;
   goal.goal.simulate = false;
   goal.goal.wait_for_execution = true;
-  gui.sendGoal(goal);
+  gui.sendGoalAndWait(goal);
```

Comment:
This code is in a for loop and with the "sendGoal" code the action server would receive the next goal without finishing the previous goal.  This has the effect of the action server only executing the last goal sent to it.  Now the action server executes each plan in the for loop as intended.